### PR TITLE
erik's Cal site updated scraper is failing

### PR DESCRIPTION
**Pull request by erikash**
@baruchiro: I've added the mapping between the `ScrapedTransaction` and `Transaction`

### CODE REVIEW

File: src/scrapers/visa-cal.ts [hunk 1]
```diff
@@ -443,38 +206,160 @@ class VisaCalScraper extends BaseScraperWithBrowser {
...
-    await this.navigateTo(TRANSACTIONS_URL, undefined, 60000);
+        const nextBillingMonth = moment().add(1, 'month');
+        const months = nextBillingMonth.diff(startMoment, 'months');
 
-    debug('fetch accounts transactions');
-    const accounts = await fetchTransactions(this.page, startMoment, this.options);
+        const allMonthsData: (CardTransactionDetails)[] = [];
+        for (let i = 0; i <= months; i += 1) {
+          const month = nextBillingMonth.clone().subtract(i, 'months');
+          const monthData = await fetchPostWithinPage<CardTransactionDetails | CardTransactionDetailsError>(
```
Comment (erikash) [inline src/scrapers/visa-cal.ts:290]: @baruchiro in what case did your get back `CardTransactionDetailsError` ? couldn't reproduce it...

Comment (baruchiro) [reply src/scrapers/visa-cal.ts:290]: It was reproduced on development, when the `Authorization` or the `X-Site-ID` are not correct.

Comment (baruchiro) [reply src/scrapers/visa-cal.ts:290]: It was reproduced on development when the `Authorization` or the `X-Site-ID` are not correct.

**Comment by esakal** [comment]
@erikash  can I close this one?

**Comment by erikash** [comment]
Yes, sure :)

**Pull request closed without merging**